formats/cbor/commonMain/src/kotlinx/serialization/cbor/Cbor.kt

```diff
@@ -27,36 +24,111 @@ import kotlinx.serialization.modules.*
  * @param ignoreUnknownKeys specifies if unknown CBOR elements should be ignored (skipped) when decoding.
+ * @param writeKeyTags Specifies whether tags set using the [KeyTags] annotation should be written (or omitted)
+ * @param writeValueTags Specifies whether tags set using the [ValueTags] annotation should be written (or omitted)
+ * @param writeObjectTags Specifies whether tags set using the [ObjectTags] annotation should be written (or omitted)
+ * @param verifyKeyTags Specifies whether tags preceding map keys (i.e. properties) should be matched against the
+ *                      [KeyTags] annotation during the deserialization process. Useful for lenient parsing
+ * @param verifyValueTags Specifies whether tags preceding values should be matched against the [ValueTags]
+ *                      annotation during the deserialization process. Useful for lenient parsing.
+ * @param verifyObjectTags Specifies whether tags preceding values should be matched against the [ObjectTags]
+ *                      annotation during the deserialization process. Useful for lenient parsing.
+ * @param alwaysUseByteString Specifies whether to always use the compact [ByteString] encoding when serializing
+ *                            or deserializing byte arrays.
+ *
+ * @param writeDefiniteLengths Specifies whether the definite length encoding should be used (as required for COSE, for example)
+ * @param preferCborLabelsOverNames Specifies whether to serialize element labels (i.e. Long from [CborLabel])
+ *                                    instead of the element names (i.e. String from [SerialName]) for map keys
  */
 @ExperimentalSerializationApi
 public sealed class Cbor(
-    internal val encodeDefaults: Boolean,
-    internal val ignoreUnknownKeys: Boolean,
+    public val encodeDefaults: Boolean,
+    public val ignoreUnknownKeys: Boolean,
+    public val writeKeyTags: Boolean,
+    public val writeValueTags: Boolean,
+    public val writeObjectTags: Boolean,
+    public val verifyKeyTags: Boolean,
+    public val verifyValueTags: Boolean,
+    public val verifyObjectTags: Boolean,
+    public val writeDefiniteLengths: Boolean,
+    public val preferCborLabelsOverNames: Boolean,
+    public val alwaysUseByteString: Boolean,
     override val serializersModule: SerializersModule
 ) : BinaryFormat {
 
     /**
-     * The default instance of [Cbor]
+     * The default instance of [Cbor]. Neither writes nor verifies tags. Uses indefinite length encoding by default.
      */
-    public companion object Default : Cbor(false, false, EmptySerializersModule())
+    public companion object Default:
+        Cbor(false, false, false, false, false, false, false, false, false, false, false, EmptySerializersModule()) {
+
+        /**
+         * Stricter instance of [Cbor]. Writes and verifies all tags. Uses indefinite length encoding by default.
+         */
+        public val Tagging: Cbor =
```

Comment:
may be better name will be `Tagged`? like `Cbor.Tagged` = `Cbor with tags`?
Not really sure.

Also, minor thing, but I believe it will be better to specify only flags that a different from defaults during construction (6 `*tags`) to bring attention what is the difference from `Cbor.Default`. No need to repeat other properties IMO

Reply:
Or it is possible to enable `writeDefiniteLengths` and rename instance to `CoseComplaint`

Reply:
I like the idea of `CodeCompliant` although, technically it still lacks sorting of map keys by name, so it might be misleading.
I considered implementing it, but if the user takes care of sorting, everything is compliant already, More importantly: I don't know how to accomplish that without a performance hit

Reply:
went a ahead and refactored it like that in e483c7f3863bc53fe85843159cbd85047c615cb7. also added info about map keys not being sorted